scripts/rollup-plugin-babel-source.js

```diff
@@ -1,21 +1,24 @@
-const path = require("path");
-const fs = require("fs");
+import path from "path";
+import fs from "fs";
+import { fileURLToPath } from "url";
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
 const dirname = path.join(__dirname, "..");
 
 const BABEL_SRC_REGEXP =
   path.sep === "/"
     ? /packages\/(babel-[^/]+)\/src\//
     : /packages\\(babel-[^\\]+)\\src\\/;
 
-module.exports = function () {
+export default function () {
   return {
     name: "babel-source",
     load(id) {
       const matches = id.match(BABEL_SRC_REGEXP);
       if (matches) {
         // check if browser field exists for this file and replace
         const packageFolder = path.join(dirname, "packages", matches[1]);
-        const packageJson = require(path.join(packageFolder, "package.json"));
+        const packageJson = import(path.join(packageFolder, "package.json"));
```

Comment:
Since JSON module is still experimental, you can create `require` from `module`: https://nodejs.org/api/esm.html#esm_commonjs_json_and_native_modules

We don't have to replace all `require` by `import` in one take. If, in this case, `require` is more convenient, go with `require`.